cmd/fastly/main.go

```diff
@@ -79,12 +78,12 @@ func main() {
 	// There are two scenarios we now want to look out for...
 	//
 	// 1. The config is using a legacy format.
-	// 2. The config is from an older CLI version.
+	// 2. The config is invalid (e.g. major version bump) for CLI version running.
 	//
 	// To prevent issues we'll replace the config with what's embedded into the CLI,
 	// as we know that is compatible with the code currently being executed.
-	if err == config.ErrLegacyConfig || file.CLI.Version != revision.SemVer(revision.AppVersion) {
-		err := file.UseStatic(cfg, config.FilePath)
+	if err == config.ErrLegacyConfig || !file.ValidConfig() {
```

Comment:
I'm wondering if we should still check explicitly for `ErrLegacyConfig`? 

The reason I'm asking is because the `ValidConfig` function will check the user's config for a `ConfigVersion` value (which in the case of a legacy config will be an empty string) and will compare it to the CLI's embedded config. 

In the case of a user having a 'legacy config', the user's config will be considered invalid and we'll use the embedded config instead. Thus the check for `ErrLegacyConfig` _before_ calling `ValidConfig()` is technically redundant.

The reason I left it in is because we still want to _return_ an `ErrLegacyConfig` type (earlier in the logic), and so it's kinda nice that we _explicitly_ highlight it as a scenario we want to be handling in this latter section of the code.